EPI/create-VS-from-CAG.py

```diff
@@ -0,0 +1,179 @@
+import json
+from argparse import ArgumentParser
+
+
+class Node:
+    def __init__(self, id):
+        self.id = id
+        self.parents = []
+        self.children = []
+
+
+def analyze_cag_edges(iden_by_ids, nodes_by_id):
+    edges_from = {}
+    for id in iden_by_ids:
+        node = nodes_by_id[id]
+        iden = iden_by_ids[id]
+        if iden not in edges_from:
+            edges_from[iden] = {}
+
+        qu = []
+        qu.append((node, 0))
+        while qu:
+            n, level = qu.pop(0)
+            for p in n.parents:
+                if p.id in iden_by_ids:
+                    from_iden, from_level = iden_by_ids[p.id], level
+                    if (from_iden not in edges_from[iden]) or \
+                            (from_iden in edges_from[iden] and from_level < edges_from[iden][from_iden]):
+                        edges_from[iden][from_iden] = from_level
+                qu.append((p, level + 1))
+    return edges_from
+
+
+def analyze_cag(input):
+    constants = set()
+    variables = set()
+    iden_by_ids = {}
+    nodes_by_id = {}
+    defs_by_idens = {}
+
+    with open(input, "rt") as f:
+        model = json.load(f)
+
+        parameters = []
+        inputs = []
+        model_variables = []
+        for metadata in model["metadata"]:
+            if metadata["type"] == "model-identifiers":
+                parameters.extend(metadata["attributes"][0]["parameters"])
+                inputs.extend(metadata["attributes"][0]["inputs"])
+                model_variables.extend(metadata["attributes"][0]["model_variables"])
+
+        for variable in model["variables"]:
+            if variable["uid"] in model_variables:
+                if "metadata" in variable:
+                    defn = variable["metadata"][0]["attributes"][0]["text_definition"]
+                    iden = variable["metadata"][0]["attributes"][0]["text_identifier"]
+                    variables.add(iden)
+                    iden_by_ids[variable["uid"]] = iden
+                    defs_by_idens[iden] = defn
+
+            nodes_by_id[variable["uid"]] = Node(variable["uid"])
+
+        for variable in model["variables"]:
```

Comment:
No, because note that `variables` is being populated in the first loop, and in the second loop, we do stuff only if a value is not in `variables`. So there is a dependency from the second loop onto the first loop.